pyomo/contrib/fme/fourier_motzkin_elimination.py

```diff
@@ -301,56 +394,144 @@ def _fourier_motzkin_elimination(self, constraints, vars_to_eliminate):
+                            self._nonneg_scalar_multiply_linear_constraint(
+                                cons, to_lcm))
 
             constraints = waiting_list
             for leq in leq_list:
                 for geq in geq_list:
-                    constraints.append(self._add_linear_constraints(leq, geq))
+                    constraints.append( self._add_linear_constraints( leq, geq))
+                    if self.verbose:
+                        cons = constraints[len(constraints)-1]
+                        logger.info("\t%s <= %s" % 
+                                    (cons['lower'], 
+                                     cons['body'].to_expression()))
 
         return constraints
 
+    def _get_noninteger_coef_error_message(self, varname, coef):
+        return ("The do_integer_arithmetic flag was "
+                "set to True, but the coefficient of "
+                "%s is non-integer within the specified "
+                "tolerance, with value %s. \n"
+                "Please set do_integer_arithmetic="
+                "False, increase integer_tolerance, "
+                "or make your data integer." % (varname, coef))
+
+    def _multiply(self, scalar, coef, error_message):
+        if self.do_integer_arithmetic:
+            if _is_integer(coef, self.integer_tolerance):
+                assert type(scalar) is int
+                return scalar*int(coef)
+            else:
+                raise ValueError(error_message)
+        elif abs(scalar*coef) > self.zero_tolerance:
+            return scalar*coef
+        else:
+            return 0
+
+    def _add(self, a, b, error_message):
+        if self.do_integer_arithmetic:
+            if _is_integer(a, self.integer_tolerance) and \
+               _is_integer(b, self.integer_tolerance):
+                return a + b
+            else:
+                raise ValueError(error_message)
+        elif abs(a + b) > self.zero_tolerance:
+            return a + b
+        else:
+            return 0
+        
     def _nonneg_scalar_multiply_linear_constraint(self, cons, scalar):
         """Multiplies all coefficients and the RHS of a >= constraint by scalar.
         There is no logic for flipping the equality, so this is just the 
         special case with a nonnegative scalar, which is all we need.
+
+        If self.do_integer_arithmetic is True, this assumes that scalar is an
+        int. It also will throw an error if any data is non-integer (within
+        tolerance)
         """
         body = cons['body']
-        body.linear_coefs = [scalar*coef for coef in body.linear_coefs]
+        new_coefs = []
+        for i, coef in enumerate(body.linear_coefs):
+            v = body.linear_vars[i]
+            new_coefs.append( 
+                self._multiply(scalar, coef,
+                               self._get_noninteger_coef_error_message(v.name,
+                                                                       coef)))
```

Comment:
This has performance implications.  The error message will always be created / formatted, even when it is not needed.  Maybe we can redesign this to pass callbacks around and only generate the messages when they are actually going to be raised?

If `_multiply` was:
```python
def _multiply(self, scalar, coef, error_message, error_args):
    #...
    raise ValueError(error_message if error_args is None
                     else error_message(*error_args))
```
then this line could be
```python
self._multiply(scalar, coef,
               self._get_noninteger_coef_error_message,
               (v.name, coef))
```
...of course, this applies to `_add` as well.